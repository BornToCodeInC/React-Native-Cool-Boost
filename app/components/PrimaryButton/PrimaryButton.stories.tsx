import React from 'react';
import {View, StyleSheet, Text} from 'react-native';
import {storiesOf} from '@storybook/react-native';
import {action} from '@storybook/addon-actions';
import {text} from '@storybook/addon-knobs';
import {PrimaryButton} from './PrimaryButton';

storiesOf('Primary Button', module)
  .addDecorator(story => <View style={styles.decorator}>{story()}</View>)
  .add('with text', () => (
    <PrimaryButton
      content={text('Button content', 'Add to cart')}
      handlePress={action('clicked-text')}
    />
  ))
  .add('with some emoji', () => (
    <PrimaryButton
      content={<Text>😀 😎 👍 💯</Text>}
      handlePress={action('clicked-text')}
    />
  ));

const styles = StyleSheet.create({
  decorator: {
    flex: 1,
    justifyContent: 'center',
  },
});
