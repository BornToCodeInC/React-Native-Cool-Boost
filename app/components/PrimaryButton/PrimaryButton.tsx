import React from 'react';
import { StyleSheet, Text, Pressable } from 'react-native';

interface Props {
    content: string | React.ReactNode;
    handlePress: () => void;
}


export const PrimaryButton: React.FC<Props> = ({ content, handlePress }): JSX.Element => {
    return (
        <Pressable
            style={styles.button}
            onPress={handlePress}
        >
            <Text style={styles.buttonText}>{content}</Text>
        </Pressable>
    );
};

const styles = StyleSheet.create({
    button: {
        width: '100%',
        backgroundColor: '#008ACE',
        paddingVertical: 15,
        borderRadius: 4,
        marginTop: 35,
    },
    buttonText: {
        textAlign: 'center',
        textTransform: 'uppercase',
        fontSize: 15,
        fontWeight: '500',
        lineHeight: 16,
        color: '#FFFFFF',
    },
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
    },
    dot: {
        position: 'absolute',
        width: 20,
        height: 20,
        backgroundColor: 'blue',
        borderRadius: 10,
    },
});
