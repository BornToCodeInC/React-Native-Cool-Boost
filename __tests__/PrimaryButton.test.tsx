import 'react-native';
import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import {PrimaryButton} from '../app/components/PrimaryButton/PrimaryButton';

describe('SearchBox.tsx', () => {
  test('renders correctly', () => {
    const handler = jest.fn();
    const {toJSON} = render(
      <PrimaryButton content="Submit" handlePress={handler} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  test('should trigger handler on button press', async () => {
    const handler = jest.fn();
    const {findByText} = render(
      <PrimaryButton content="Submit" handlePress={handler} />,
    );
    const btn = await findByText('Submit');
    fireEvent.press(btn);
    expect(handler).toHaveBeenCalled();
  });
});
