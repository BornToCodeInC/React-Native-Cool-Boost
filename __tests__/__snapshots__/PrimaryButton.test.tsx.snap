// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox.tsx renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#008ACE",
      "borderRadius": 4,
      "marginTop": 35,
      "paddingVertical": 15,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": 15,
        "fontWeight": "500",
        "lineHeight": 16,
        "textAlign": "center",
        "textTransform": "uppercase",
      }
    }
  >
    Submit
  </Text>
</View>
`;
